src/lightning/pytorch/trainer/setup.py

```diff
@@ -178,11 +182,24 @@ def _log_device_info(trainer: "pl.Trainer") -> None:
         hpu_available = False
     rank_zero_info(f"HPU available: {hpu_available}, using: {num_hpus} HPUs")
 
+    if _LIGHTNING_XPU_AVAILABLE:
+        from lightning_xpu.pytorch import XPUAccelerator
+
+        num_xpus = trainer.num_devices if isinstance(trainer.accelerator, XPUAccelerator) else 0
+        xpu_available = XPUAccelerator.is_available()
+    else:
+        num_xpus = 0
+        xpu_available = False
+    rank_zero_info(f"XPU available: {xpu_available}, using: {num_xpus} XPUs")
```

Comment:
it seems that num_xpus and xpu_available are local vars,  line 193 shall report error here.

Reply:
they are set in either if or else scopes.

Reply:
> they are set in either if or else scopes.

yes, but that means local var